internal/syncd/syncd.go

```diff
@@ -0,0 +1,89 @@
+package syncd
+
+import (
+	"context"
+	"sync"
+
+	"ship-it/internal"
+
+	"k8s.io/helm/pkg/proto/hapi/chart"
+)
+
+// ImageReconciler is a callback that reconciles a new image with some external
+// state. For example, updating the values of a helm chart in a remote
+// repository to use the new image's tag.
+type ImageReconciler interface {
+	Reconcile(context.Context, *internal.Image) error
+}
+
+// ImageListener models a stream of `*Image` events. It calls the
+// ImageReconciler to reconcile each new image with some external state.
+type ImageListener interface {
+	Listen(context.Context, ImageReconciler) error
+}
+
+// ChartReconciler is an callback that reconciles a new chart with some external
+// state. For example, deploying the chart to a kubernetes cluster.
+type ChartReconciler interface {
+	Reconcile(context.Context, *chart.Chart) error
+}
+
+// ChartListener models a stream of `*chart.Chart` events. It calls the
+// ChartReconciler to reconcile each new chart with some external state.
+type ChartListener interface {
```

Comment:
Is this the thing that listens for changes to the registry chart?

Reply:
Yes, it listens for changes to the registry chart. But the syncd pkg doesn't need to know or care that it's for a specific chart. Here's how we would implement it with github+sqs: https://github.com/Wattpad/ship-it/pull/50/files#diff-4fd52b3f4241e7dea56c262b4ce0bc67

Reply:
I feel like this abstraction is going slightly too far. The job of this sync component is always going to be to reconcile between resources in the service registry, images in a docker repository, and the state of the cluster ship-it is deployed to. 

I would prefer that here we explicitly called this `RegistryListener` or `RegistryChartListener` to make the responsibility of this component more clear. I get that we're trying to be abstract to make this pluggable, but I think the tool can prescribe that the registry is going to be a chart.

Reply:
`RegistryChartListener` sounds good to me. I updated the docs as well to be less vague about what these interfaces are meant for